src/category_theory/FinType.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import data.fintype.basic
+import data.fin
+import category_theory.concrete_category.bundled
+import category_theory.concrete_category
+import category_theory.full_subcategory
+import category_theory.skeletal
+
+/-!
+# The category of finite types.
+
+We define the category of finite types, denoted `FinType` as
+(bundled) types with a `fintype` instance.
+
+We also define `FinType.Skeleton`, the standard skeleton of `FinType` whose objects are `fin n`
+for `n : ℕ`. We prove that the obvious inclusion functor `FinType.Skeleton ⥤ FinType` is an
+equivalence of categories in `FinType.Skeleton.equivalence`.
+-/
+
+open_locale classical
+open category_theory
+
+/-- The category of finite types. -/
+@[derive has_coe_to_sort]
+def FinType := bundled fintype
+
+namespace FinType
+
+/-- Construct a bundled `FinType` from the underlying type and typeclass. -/
+def of (X : Type*) [fintype X] : FinType := bundled.of X
+instance : inhabited FinType := ⟨⟨pempty⟩⟩
+instance {X : FinType} : fintype X := X.2
+
+instance : category FinType := induced_category.category bundled.α
+
+/-- The fully faithful embedding of `FinType` into the category of types. -/
+@[derive [full, faithful]]
+def incl : FinType ⥤ Type* := induced_functor _
```

Comment:
We might want some simp lemmas for this, possibly using `simps rhs_md` stuff

Reply:
What sort of simp lemmas would you like to have? I can't seem to make `@[simps ...]` work automatically...

Reply:
Does `@[simps {rhs_md:=semireducible}]` not work? If not, just `rfl` lemmas for `incl_obj` and `incl_map` seem like the sensible ones here.

Reply:
Thanks, that works. Added it.